front/components/DataSourceResourceSelectorTree.tsx

```diff
@@ -85,14 +89,38 @@ function DataSourceResourceSelectorChildren({
   isChecked: boolean;
   selectedParentIds: Set<string>;
   onSelectChange: (
-    resource: { resourceId: string; resourceName: string },
+    resource: { resourceId: string; resourceName: string; parents: string[] },
     selected: boolean
   ) => void;
+  parentsById: Record<string, Set<string>>;
 }) {
   const { resources, isResourcesLoading, isResourcesError } =
-    useConnectorPermissions(owner, dataSource, parentId, null);
+    useConnectorPermissions({
```

Comment:
So we retrieve parents at each expand?

Reply:
We retrieve parents when ever the SWR reloads. For now this page doesn't support "edit"

Reply:
Right but we want  to have the story for edition figured out right now no?